src/main/java/codeu/service/BBCodeParser.java

```diff
@@ -0,0 +1,29 @@
+package codeu.service;
+
+public class BBCodeParser {
+
+  /**
+   * Parses the input, substituting BBCode tags with their equivalent HTML tags
+   * @param input The text to be parsed
+   * @return The parsed text
+   */
+  public static String parse(String input) {
+    StringBuilder builder = new StringBuilder(input);
+
+    // replace [b] with <b>
+    int i = builder.indexOf("[b]");
+    while (i != -1) {
+      builder.replace(i, i + "[b]".length(), "<b>");
+      i = builder.indexOf("[b]", i);
+    }
+
+    // replace [/b] with </b>
+    i = builder.indexOf("[/b]");
```

Comment:
Same as above.

Reply:
Also, I was thinking about doing one run through the string and comparing against all the possible tags, but I think that would end up being the same work as making multiple runs but comparing against one tag, right? (So it would either be one loop with many operations per iteration or many loops with one operation per iteration).